site/app/templates/officeHoursQueue/QueueStatus.twig

```diff
@@ -190,8 +201,86 @@
   </div>
 {% endif %}
 <script>
+
+    if(localStorage.getItem('QueueMessage')){
+        let current_time = new Date();
+        const sent = JSON.parse(localStorage.getItem('QueueMessageExpire'));
+        const queue = sent.queue;
+        const sent_time = sent.time;
+        const course = sent.course;
+        //get rid of message after 80 minutes
+        if(current_time.getTime() - 5000000 < sent_time) {
+            if(course === '{{ viewer.getCourse() }}'){
+                $('.queue-message').css('display', 'block');
+                $('#queue-message-msg').html(localStorage.getItem('QueueMessage'));
+                $('#queue-message-msg-time').html(sent.formatted_time);
+            }
+        } else {
+            localStorage.removeItem('QueueMessageExpire');
+            localStorage.removeItem('QueueMessage');
+        }
+
+    }
+
+     {% if viewer.inQueue() %}
+        function queueMessage(message, queue, course) {
+            if (queue === "{{viewer.getCurrentQueueCode()}}"){
+                let now = new Date();
+                const item = {
+                    queue: queue,
+                    time: now.getTime(),
+                    formatted_time: now.toLocaleString(),
+                    course: course
+                }
+                localStorage.setItem('QueueMessageExpire', JSON.stringify(item));
+                localStorage.setItem('QueueMessage', message);
+                $('.queue-message').css('display', 'block');
+                $('#queue-message-msg').html(message);
+                $('#queue-message-msg-time').html(now.toLocaleString());
```

Comment:
By setting raw html here and in 215-216, this could opens a XSS vulnerability by setting raw script tags. I tested it and was able to run some random javascript code.